clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -98,13 +116,20 @@ private static ProduceResponseData toData(Map<TopicPartition, PartitionResponse>
                     .setLogAppendTimeMs(response.logAppendTime)
                     .setErrorMessage(response.errorMessage)
                     .setErrorCode(response.error.code())
+                    .setCurrentLeader(response.currentLeader != null ? response.currentLeader : new LeaderIdAndEpoch())
```

Comment:
do we need to set anything here if the response is null?

Reply:
Or alternatively pass in the default and not have to do a check here.

Reply:
based on the change I made in handleProduceRequest I don't think currentLeader can be null anymore, I removed this check